parlai/agents/transformer/modules.py

```diff
@@ -310,17 +349,23 @@ def forward(self, input):
         if self.embeddings_scale:
             tensor = tensor * np.sqrt(self.dim)
         tensor = tensor + self.position_embeddings(positions).expand_as(tensor)
+        if self.embedding_normalization:
+            tensor = self.layer_norm_emb(tensor)
```

Comment:
So this was a bit of a surprising change to me, because it differs from what fairseq/tensor2tensor do (and what I THOUGHT you were implementing):

https://github.com/pytorch/fairseq/blob/b26d6b58b0bd55c1b59aa89f1b03d79011dbe7aa/fairseq/models/transformer.py#L604-L618

That is, the layer normalizations either happen before the attn/ffn (tensor2tensor mode), or they happen after the residuals (aiayn mode). You're saying there's a third version used in XLM? Where there's just one extra LN around the embeddings, but otherwise it's normal? (cc @glample).

Reply:
I do not use the fairseq implementation but the XLM implemntation ( same as ?BERT) which is here : https://github.com/facebookresearch/XLM

Reply:
Good, thanks for clarifying that.

I still would like to have some sort of `--variant {'aiayn', 'xlm'}` arg instead of `--embedding-normalization`